starknet_py/tests/e2e/client/client_test.py

```diff
@@ -204,31 +204,58 @@ async def test_get_class_by_hash(client, class_hash):
 
 
 @pytest.mark.asyncio
-async def test_wait_for_tx_accepted(gateway_client):
+@pytest.mark.parametrize(
+    "client, get_tx_receipt",
+    [
+        (
+            "full_node_client",
+            "starknet_py.net.full_node_client.FullNodeClient.get_transaction_receipt",
+        ),
+        (
+            "gateway_client",
+            "starknet_py.net.gateway_client.GatewayClient.get_transaction_receipt",
+        ),
+    ],
+)
+async def test_wait_for_tx_accepted(client, get_tx_receipt, request):
     with patch(
-        "starknet_py.net.gateway_client.GatewayClient.get_transaction_receipt",
+        get_tx_receipt,
         AsyncMock(),
     ) as mocked_receipt:
         mocked_receipt.return_value = TransactionReceipt(
             hash=0x1, status=TransactionStatus.ACCEPTED_ON_L2, block_number=1
         )
-
-        block_number, tx_status = await gateway_client.wait_for_tx(tx_hash=0x1)
+        client = request.getfixturevalue(client)
```

Comment:
Why was this added?

Reply:
Because we don't want to mix GatewayClient methods with FullNodeClient instance (the topic you mentioned above this thread).